pyuvsim/source.py

```diff
@@ -145,48 +189,45 @@ def alt_az_calc(self, time, telescope_location):
 
-        source_altaz = self.skycoord.transform_to(AltAz(obstime=time, location=telescope_location))
+        if self.time == time:  # Don't repeat calculations
+            return
 
-        alt_az = (source_altaz.alt.rad, source_altaz.az.rad)
-        self.time = time
-        self.alt_az = alt_az
-        return alt_az
+        skycoord_use = SkyCoord(self.ra, self.dec, frame='icrs')
+        source_altaz = skycoord_use.transform_to(AltAz(obstime=time, location=telescope_location))
 
-    def get_alt_az(self, time, telescope_location):
-        """ Reuse alt_az if already calculated """
-        if (self.alt_az is None) or (not time == self.time):
-            return self.alt_az_calc(time, telescope_location)
-        else:
-            return self.alt_az
+        time.location = telescope_location
+        lst = time.sidereal_time('apparent')
 
-    def pos_lmn(self, time, telescope_location):
-        """
-        calculate the direction cosines of this source at a time & location
+        cirs_source_coord = skycoord_use.transform_to('cirs')
+        tee_ra = simutils.cirs_to_tee_ra(cirs_source_coord.ra, time)
 
-        Args:
-            time: astropy Time object
-            telescope_location: astropy EarthLocation object
+        self.hour_angle = None
+        self.hour_angle = (lst - tee_ra).rad
 
-        Returns:
-            (l, m, n) direction cosine values
-        """
-        # calculate direction cosines of source at current time and array location
-        # Will only do the calculation if time has changed
-        alt_az = self.get_alt_az(time, telescope_location)
+        self.time = time
+        alt_az = np.array([source_altaz.alt.rad, source_altaz.az.rad])
 
-        # Horizon Mask
-        if alt_az[0] < 0:
-            return None
+        self.alt_az = alt_az
+
+        pos_l = np.sin(alt_az[1, :]) * np.cos(alt_az[0, :])
+        pos_m = np.cos(alt_az[1, :]) * np.cos(alt_az[0, :])
+        pos_n = np.sin(alt_az[0, :])
 
-        pos_l = np.sin(alt_az[1]) * np.cos(alt_az[0])
-        pos_m = np.cos(alt_az[1]) * np.cos(alt_az[0])
-        pos_n = np.sin(alt_az[0])
+        self.pos_lmn[0, :] = pos_l
+        self.pos_lmn[1, :] = pos_m
+        self.pos_lmn[2, :] = pos_n
 
-        return (pos_l, pos_m, pos_n)
+        # Horizon mask:
+        self.horizon_mask = self.alt_az[0, :] < 0.0
 
     def __eq__(self, other):
-        return (np.isclose(self.ra.deg, other.ra.deg, atol=self.pos_tol)
-                and np.isclose(self.dec.deg, other.dec.deg, atol=self.pos_tol)
-                and np.all(self.stokes == other.stokes)
-                and (self.name == other.name)
-                and (self.freq == other.freq))
+        return (np.allclose(self.ra.deg, other.ra.deg, atol=self.pos_tol)
+                and np.allclose(self.stokes, other.stokes)
+                and np.all(self.name == other.name)
+                and np.all(self.freq == other.freq))
```

Comment:
I notice that the time isn't included here, so the objects can be equal if the times are different. I find that confusing because the methods would give different results in that case.

Reply:
I think the `__eq__` method was written before `time` was an attribute. I'll add it here.